src/index.js

```diff
@@ -1,363 +1,136 @@
-    db.query(selectImage, [params.fileName,
-      params.resolutionX,
-      params.resolutionY,
-      params.fit,
-      parsing.supportedFileType(params.fileType)
-    ], (err, data) => {
-      if (!err && data.rowCount === 1) {
-        // It is in the cache, so redirect to there
-        helpers.send307DueToCache(res, params, data.rows[0].url);
-        return;
-      }
-
-      // It does not exist in the cache, so generate and upload
-      Image.encodeAndUpload(params, res);
-    });
-  },
-  encodeAndUpload(params, res) {
-    const file = `${config.get('originals_dir')}/${params.fileName}`;
-    fs.access(file, (err) => {
-      if (err) {
-        log.log('warn', `File ${params.fileName} was requested but did not exist`);
-        helpers.send404(res, params.fileName);
-        return;
-      }
-
-      // Get the image and resize it
-      res.writeHead(200, {
-        'Content-Type': parsing.supportedFileType(params.fileType),
-        Expires: helpers.farFutureDate(),
-        'Cache-Control': 'public'
-      });
+process.on('uncaughtException', function (err) {
+  console.log(err);
+});
 
-      // These files have already been oriented!
-      correctlyResize(file, params, (resized) => {
-        resized.stream(params.fileType, (err, stdout) => {
-          const r = stdout.pipe(res);
-          r.on('finish', () => {
-            // This is to close the result while a background job will continue to process
-            log.log('info', 'Finished sending a converted image');
-            res.end();
-          });
-        });
-      });
+const promiseUpload = (form, request) => {
+  return new Promise((resolve, reject) => form.parse(request, (err, fields, files) => err ? reject(err) : resolve(files)));
+};
 
-      correctlyResize(file, params, (resized) => {
-        resized.toBuffer(params.fileType, (err, stream) => {
-          if (!err) {
-            // This might mean we have generated the same file while an upload was in progress.
-            // However this is still better than not being able to server the image
-            Image.uploadToCache(params, stream);
-          }
-        });
-      });
-    });
-  },
-  uploadToCache(params, content) {
-    // Upload to AWS
-    const key = getKeyFromParams(params);
-    // AWS sets the etag as MD5 of the file already
-    const uploadParams = {
-      Bucket: config.get('aws.bucket'),
-      Key: key,
-      ACL: 'public-read',
-      Body: content,
-      Expires: helpers.farFutureDate(),
-      ContentType: parsing.supportedFileType(params.fileType),
-      // We let any intermediate server cache this result as well
-      CacheControl: 'public'
+const cropParametersOnUpload = (req) => {
+  const xOffset = Number(req.query.x) || undefined;
+  const yOffset = Number(req.query.y) || undefined;
+  const width = Number(req.query.width) || undefined;
+  const height = Number(req.query.height) || undefined;
+  if (xOffset && yOffset && width && height) {
```

Comment:
oof, this won't work when you provide `0` for any of these, which is very plausible